### PR TITLE
Feat: add static-file serving e2e tests

### LINKED ISSUE #114: Static File Serving E2E test
**Tests:**
- Serving files with correct MIME types
- Range requests (partial content)
- ETag and conditional requests
- 404 for missing files
- Directory traversal prevention
- Dotfile handling

### PR DESCRIPTION
**Pull request by VikramAditya33**
Closes #114 
**Tests added:**
- Serving files with correct MIME types
- Range requests (partial content)
- ETag and conditional requests
- 404 for missing files
- Directory traversal prevention
- Dotfile handling
- A bit more

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive end-to-end test suite for static file serving functionality, ensuring robust validation of MIME types, HTTP caching behavior, byte-range request handling, path traversal protection, conditional request support, proper URL encoding handling, correct route precedence when combined with API routes, dotfile access controls, and symlink security restrictions across various configuration scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/FOSSFORGE/uWestJS/pull/115)

### CODE REVIEW

File: test/http/static-file-serving.e2e.spec.ts [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+    });
+  });
+
+  // ============================================================================
+  // Edge Case Paths
+  // ============================================================================
+
+  describe('edge case paths', () => {
+    it('should handle URL-encoded spaces in path', async () => {
+      fs.writeFileSync(path.join(tempDir, 'space test.txt'), 'space content');
```
[the remark this answers is not in the document]
Comment (VikramAditya33) [reply test/http/static-file-serving.e2e.spec.ts]: alright see this one makes sense. check now
